src/exec.h

```diff
@@ -43,40 +17,58 @@ struct ExecError : std::runtime_error {
 };
 
 template <typename... Args>
-static void exec(const std::string& cmd, const std::string& err_msg_prefix, Args&&... args) {
-  // Implementation is based on test_utils.cc:Process::spawn that has been proven over time
-  std::future<std::string> err_output;
-  std::future<int> child_process_exit_code;
-  boost::asio::io_context io_context;
+static void exec(const std::string& cmd, const std::string& err_msg_prefix,
+                 const boost::filesystem::path& start_dir = "", std::string* output = nullptr) {
+  std::string command = "timeout 900 " + cmd + " 2>&1";
+  if (!start_dir.empty()) {
+    command = "cd " + start_dir.string() + " && " + command;
+  }
 
+  LOG_DEBUG << "Running: `" << command << "`";
+  FILE* pipe = popen(command.c_str(), "r");
+  if (!pipe) {
+    throw std::runtime_error("exec: popen() failed!");
+  }
+  std::string err_msg;
   try {
-    LOG_DEBUG << "Running: `" << cmd << "`";
-    bp::child child_process(cmd, bp::std_err > err_output, bp::on_exit = child_process_exit_code, io_context,
-                            std::forward<Args>(args)...);
-
-    io_context.run();
+    std::string result;
```

Comment:
Maybe we can declare `result` before the `try`, in this case the `err_msg` can be set in the `if (exit_code != EXIT_SUCCESS)` scope.

Reply:
I've removed the `err_msg` variable, as it was redundant with `result`, and adjusted the scope of the `result` variable.